presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -1846,7 +1849,7 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         checkArgument(!handle.getAnalyzePartitionValues().isPresent() || constraint.getSummary().isAll(), "Analyze should not have a constraint");
 
         HivePartitionResult partitionResult = partitionManager.getPartitions(metastore, new HiveIdentity(session), handle, constraint);
```

Comment:
Do you think it'd be better to store the set of referenced columns in HivePartitionResult and not add a new argument to partitionManager.applyPartitionResult method?

Reply:
Actually, both options kind of suck. The way I'd do it is by introducing a builder class and do something like:

```
HiveTableHandle newHandle = HiveTableHandle.builderFrom(handle)
    .withColumns(constraint.getColumns())
    .build()
```

The same mechanism would be used in `applyPartitionResult` to copy the original handle and override a few fields.

But don't worry about it. It's something we can address later. Let's keep the argument in `applyPartitionResult` for now.